artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -2357,11 +2357,19 @@ public synchronized RoutingStatus doSend(final Transaction tx,
          routingContext.setAddress(art.getName());
          routingContext.setRoutingType(art.getRoutingType());
 
+         // Retrieve message size for metrics update before routing,
+         // since large message backing files may be closed once routing completes
+         int lmSize = 0;
```

Comment:
variable name needs updated (here and in updateProducerMetrics), lmSize doesnt make sense now

Reply:
ha, yes. Thanks Robbie. I think I can also use ?: form to combine the statements into one line.

Reply:
Done!